tests/queries/0_stateless/02381_rocksdb_with_ttl.sql

```diff
@@ -0,0 +1,32 @@
+DROP TABLE IF EXISTS 02381_test;
+
+CREATE TABLE 02381_test (key UInt64, value String) Engine=EmbeddedRocksDB(10) PRIMARY KEY(key);
+
+SELECT value FROM system.rocksdb WHERE database = currentDatabase() and table = '02381_test' and name = 'number.keys.written';
+INSERT INTO 02381_test SELECT number, format('Hello, world ({})', toString(number)) FROM numbers(10000);
+SELECT value FROM system.rocksdb WHERE database = currentDatabase() and table = '02381_test' and name = 'number.keys.written';
+
+SELECT * FROM 02381_test WHERE key = 123;
+SELECT '--';
+SELECT * FROM 02381_test WHERE key = -123;
+SELECT '--';
+SELECT * FROM 02381_test WHERE key = 123 OR key = 4567 ORDER BY key;
+SELECT '--';
+SELECT * FROM 02381_test WHERE key = NULL;
+SELECT '--';
+SELECT * FROM 02381_test WHERE key = NULL OR key = 0;
+SELECT '--';
+SELECT * FROM 02381_test WHERE key IN (123, 456, -123) ORDER BY key;
+SELECT '--';
+SELECT * FROM 02381_test WHERE key = 'Hello'; -- { serverError 53 }
+
+DETACH TABLE 02381_test NO DELAY;
+ATTACH TABLE 02381_test;
+
+SELECT * FROM 02381_test WHERE key IN (99, 999, 9999, -123) ORDER BY key;
+
+sleep(10)
```

Comment:
this could become flaky

also, now I don't see a reliable way to test ttl because the values are removed during compaction, not something in our control...
Maybe we can remove the TTL test and just leave the read_only.
What do you think?

Reply:
Maybe it worth adding [`CompactRange`](https://github.com/facebook/rocksdb/wiki/Manual-Compaction) support, i.e. implement it as `IStorage::optimize`, and also `IStorage::flush` can be implemented as [`DB::Flush`](https://github.com/facebook/rocksdb/wiki/Atomic-flush)

And later this can be used instead of `sleep`

Reply:
I agree. But there are some error in the code, I will fix them after work.